users/src/main/java/io/crate/http/netty/HttpAuthUpstreamHandler.java

```diff
@@ -0,0 +1,120 @@
+/*
+ * This file is part of a module with proprietary Enterprise Features.
+ *
+ * Licensed to Crate.io Inc. ("Crate.io") under one or more contributor
+ * license agreements.  See the NOTICE file distributed with this work for
+ * additional information regarding copyright ownership.
+ *
+ * Unauthorized copying of this file, via any medium is strictly prohibited.
+ *
+ * To use this file, Crate.io must have given you permission to enable and
+ * use such Enterprise Features and you must have a valid Enterprise or
+ * Subscription Agreement with Crate.io.  If you enable or use the Enterprise
+ * Features, you represent and warrant that you have a valid Enterprise or
+ * Subscription Agreement with Crate.io.  Your use of the Enterprise Features
+ * if governed by the terms and conditions of your Enterprise or Subscription
+ * Agreement with Crate.io.
+ */
+
+package io.crate.http.netty;
+
+import com.google.common.annotations.VisibleForTesting;
+import io.crate.operation.auth.*;
+import org.apache.logging.log4j.Logger;
+import org.elasticsearch.common.logging.Loggers;
+import org.jboss.netty.channel.*;
+import org.jboss.netty.handler.codec.http.*;
+import org.jboss.netty.util.CharsetUtil;
+
+import javax.annotation.Nullable;
+import java.net.InetAddress;
+import java.util.Locale;
+
+import static org.jboss.netty.buffer.ChannelBuffers.*;
+
+public class HttpAuthUpstreamHandler extends SimpleChannelUpstreamHandler {
+
+    private static final Logger LOGGER = Loggers.getLogger(HttpAuthUpstreamHandler.class);
+    private final Authentication authService;
+    private boolean authorized;
+
+    public HttpAuthUpstreamHandler(Authentication authService) {
+        super();
+        this.authService = authService;
+    }
+
+    @Override
+    public void messageReceived(ChannelHandlerContext ctx, MessageEvent e) throws Exception {
+        Object msg = e.getMessage();
+        if (msg instanceof HttpRequest) {
+            HttpRequest request = (HttpRequest) msg;
+            String username = request.headers().contains(AuthenticationProvider.HTTP_HEADER_USER) ?
+                request.headers().get(AuthenticationProvider.HTTP_HEADER_USER) :
+                TrustAuthentication.SUPERUSER;
```

Comment:
Isn't there a setting that allows to overwrite the default user that should be used?